src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -154,7 +162,7 @@ public PerLeafResult searchLeaf(LeafReaderContext context, int k) throws IOExcep
             Map<Integer, Float> result = doExactSearch(context, docs, cardinality, k);
             return new PerLeafResult(filterWeight == null ? null : filterBitSet, result);
         }
-        return new PerLeafResult(filterWeight == null ? null : filterBitSet, docIdsToScoreMap);
+        return new PerLeafResult((filterWeight == null || cardinality == maxDoc) ? null : filterBitSet, docIdsToScoreMap);
```

Comment:
Why do we have to make this change here? I believe updating filter to null is only required for AnnSearch. is there a reason why we have to update the return result as well?